contracts/src/SubstrateTypes.sol

```diff
@@ -196,4 +205,64 @@ library SubstrateTypes {
             ScaleCodec.encodeU128(xcmFee)
         );
     }
+
+    // Encode V2 Payload
+    //
+    // ```rust
+    // struct Payload {
+    //   origin: H160,
+    //   assets: Vec<Asset>
+    //   xcm: Vec<u8>
+    // }
+    //
+    function encodePayloadV2(address origin, bytes[] memory assets, bytes memory xcm)
+        internal
+        pure
+        returns (bytes memory)
+    {
+        return bytes.concat(abi.encodePacked(origin), VecAsset(assets), VecU8(xcm));
```

Comment:
Is `assets` required here?  I would assume On BH we just need the encoded xcm and forward to AH transparently?

Reply:
Yes the on-chain code needs to know what assets are being transferred so that it can lock/burn them. I added more documentation on this in https://github.com/Snowfork/snowbridge/pull/1300/commits/149c3e425bbcaf023a9564977921bd355cf129fd.